### PR TITLE
Feature/CLI

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#13

### CODE REVIEW

File: reserver/reserver_param.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
 # -*- coding: utf-8 -*-
 """Parameters and constants."""
+OVERVIEW = """
+A Python that offers the ability to quickly reserve a PyPI package name.
```
Comment (sepandhaghighi) [inline reserver/reserver_param.py:4]: Wrong overview.

`Reserver is an open source Python package that offers the ability to quickly reserve a PyPI package name. Got a notion? Before it's taken, immediately reserve the product name!`

File: reserver/__main__.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        nargs='+',
+        help='Name(s) to get reserved',
+        required=True
+    )
+    parser.add_argument(
+        '--token',
+        help='The token for (main|test) PyPI account',
+        required=True
+    )
+    parser.add_argument('--test', action='store_true', help='Flag identifying test account or not')
```
Comment (sepandhaghighi) [inline reserver/__main__.py:28]: My suggestion:

`help='test PyPI (test.pypi.org)`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 💯 
Please take a look at my comments 🥇

File: reserver/reserver_param.py (hunk 1 above)
Comment (sepandhaghighi) [reply reserver/reserver_param.py:4]: Or

`A Python package that offers the ability to quickly reserve a PyPI package name. Got a notion? Before it's taken, immediately reserve the product name!`

**Comment by AHReccese** [comment]
@sepandhaghighi 
A kind reminder for the review!
Thanks!

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 💯 

Please take a look at my comments 🔥 

Also:

1. Don't forget to add `entry_points` to `setup.py`

```
entry_points={
            'console_scripts': [
                'reserver = reserver.__main__:main',
            ]
    }
```
2. Add a smoke test (first test) for CLI to workflow. Something like this:

```console
reserver --version
```

File: reserver/__main__.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        help='The token for (main|test) PyPI account',
+        required=True
+    )
+    parser.add_argument('--test', action='store_true', help='test PyPI (test.pypi.org)')
+    parser.add_argument('--version', help="version", action='store_true', default=False)
+    parser.add_argument('-v', help="version", action='store_true', default=False)
+    args = parser.parse_known_args()[0]
+    if args.version or args.v:
+        print(RESERVER_VERSION)
+    names = args.name
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/__main__.py:34]: Which one do you suggest to get done in this command? version check or package reserving?
@sepandhaghighi

File: reserver/__main__.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    CLI main function.
+
+    :return: None
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--name',
+        nargs='+',
+        help='Name(s) to get reserved',
+        required=True
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/__main__.py]: Fixed.

File: reserver/reserver_func.py [hunk 5]
```diff
@@ -48,7 +50,32 @@ def does_package_exist(suggested_name, test_pypi):
...
+    :type user_parameters: dict
+    :param regex: name of the regex to get applied
+    :type regex: str
+    :return: value of the associated parameter
+    """
+    if not user_parameters or parameter not in user_parameters:
+        if parameter in PACKAGE_PARAMETERS:
+            return PACKAGE_PARAMETERS[parameter]
+        else:
+            raise ReserverBaseError("Given parameter doesn't exist among the supported user allowed parameters.")
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/reserver_func.py]: Done.

File: reserver/reserver_func.py [hunk 6]
```diff
@@ -48,7 +50,32 @@ def does_package_exist(suggested_name, test_pypi):
...
+    if regex:
+        if re.match(VALIDATIONS[regex], user_parameters[parameter]):
+            return user_parameters[parameter]
+        else:
+            raise ReserverBaseError("Invalid value for " + parameter + " that should be a valid " + regex)
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/reserver_func.py]: Done.

File: reserver/__main__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/__main__.py:37]: Sepand -> --version
Sadra -> --version
Amir -> --version

so -> --version

[the remark this answers is not in the document]
Comment (AHReccese) [reply reserver/__main__.py:37]: Applied.

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🚀

**Pull request merged**